src/docs/theia_ai.md

```diff
@@ -0,0 +1,404 @@
+
+<img src="../../used-variables-command-agent.png" alt="Used variables in Theia AI" style="max-width: 525px">
+
+To declare an agent-specific variable, simply add it like this in the constructor of your agents, along with an optional description and whether it is used in the prompt template or not.
+
+
+```ts
+this.agentSpecificVariables = [{
+           name: 'command-ids',
+           description: 'The list of available commands in Theia.',
+           usedInPrompt: true
+       }];
+```
+
+#### Global Variables
+
+Global variables are available to all agents. Therefore, they are provided as a dedicated contribution. In the following, we will define a simple global variable that resolves to the current date, i.e. to the time when the variable get resolved in a request. You can review the [full code of this variable](https://github.com/eclipse-theia/theia/blob/master/packages/ai-core/src/common/today-variable-contribution.ts) for a complete description.
+A Global variable basically consists of four parts.
+
+First, we define the variable, as it can be used in requests. In the following example, we define two optional arguments to influence the format.
+
+```ts
+export const TODAY_VARIABLE: AIVariable = {
+   id: 'today-provider',
+   description: 'Does something for today',
+   name: 'today',
+   args: [
+       { name: TodayVariableArgs.IN_ISO_8601, description: 'Returns the current date in ISO 8601 format' },
+       { name: TodayVariableArgs.IN_UNIX_SECONDS, description: 'Returns the current date in unix seconds format' }
+   ]
+};
+```
+
+Second, we register the variable using an ‘AIVariableContribution’ (see code example below). Third, we resolve the variable (also see code example below). Please note that for resolving a variable, you can access all necessary APIs, including tool-specific API or custom services.
+
+```ts
+export class TodayVariableContribution implements AIVariableContribution, AIVariableResolver {
+   /Register the variable resolver
+   registerVariables(service: AIVariableService): void {
+       service.registerResolver(TODAY_VARIABLE, this);
+   }
+
+
+
+
+   async resolve(request: AIVariableResolutionRequest, context: AIVariableContext): Promise<ResolvedAIVariable | undefined> {
+       if (request.variable.name === TODAY_VARIABLE.name) {
+           return this.resolveTodayVariable(request);
+       }
+       return undefined;
+   }
+
+
+   private resolveTodayVariable(request: AIVariableResolutionRequest): ResolvedTodayVariable {
+       const date = new Date();
+       if (request.arg === TodayVariableArgs.IN_ISO_8601) {
+           return { variable: request.variable, value: date.toISOString(), date };
+       }
+       if (request.arg === TodayVariableArgs.IN_UNIX_SECONDS) {
+           return { variable: request.variable, value: Math.round(date.getTime() / 1000).toString(), date };
+       }
+       return { variable: request.variable, value: date.toDateString(), date };
+   }
+}
+```
```

Comment:
```suggestion
```ts
export class TodayVariableContribution implements AIVariableContribution, AIVariableResolver {
   // Register the variable resolver
   registerVariables(service: AIVariableService): void {
       service.registerResolver(TODAY_VARIABLE, this);
   }

   async resolve(request: AIVariableResolutionRequest, context: AIVariableContext): Promise<ResolvedAIVariable | undefined> {
       if (request.variable.name === TODAY_VARIABLE.name) {
           return this.resolveTodayVariable(request);
       }
       return undefined;
   }

   private resolveTodayVariable(request: AIVariableResolutionRequest): ResolvedTodayVariable {
       const date = new Date();
       if (request.arg === TodayVariableArgs.IN_ISO_8601) {
           return { variable: request.variable, value: date.toISOString(), date };
       }
       if (request.arg === TodayVariableArgs.IN_UNIX_SECONDS) {
           return { variable: request.variable, value: Math.round(date.getTime() / 1000).toString(), date };
       }
       return { variable: request.variable, value: date.toDateString(), date };
   }
}
```
```